src/main/java/Iris/ToDoList.java

```diff
@@ -0,0 +1,157 @@
+package iris;
+
+import java.util.ArrayList;
+
+/**
+ * Represents a list of tasks in the Iris application.
+ */
+public class ToDoList {
+    private final ArrayList<Task> list;
+
+    /**
+     * Constructor for the ToDoList class.
+     *
+     * @param list The initial list of tasks.
+     */
+    public ToDoList(ArrayList<Task> list) {
+        this.list = list;
+    }
+
+    /**
+     * Adds a task to the list.
+     *
+     * @param task The task to be added.
+     */
+    public void add(Task task) {
+        list.add(task);
+    }
+
+    /**
+     * Removes a task from the list by index.
+     *
+     * @param index The index of the task to be removed.
+     * @throws IndexOutOfBoundsException If the index is out of range.
+     */
+    public void remove(int index) {
+        if (index < 1 || index > list.size()) {
+            throw new IndexOutOfBoundsException("Task index is out of range.");
+        }
+        list.remove(index - 1);
+    }
+
+    /**
+     * Gets a task from the list by index.
+     *
+     * @param index The index of the task to be retrieved.
+     * @return The task at the specified index.
+     * @throws IndexOutOfBoundsException If the index is out of range.
+     */
+    public Task get(int index) {
+        if (index < 1 || index > list.size()) {
```

Comment:
I see this line is repeated multiple times across many functions. Perhaps can abstract this out to reduce duplicated codes.